pkgs/games/abuse/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, fetchurl, makeDesktopItem, copyDesktopItems, SDL, SDL_mixer, freepats }:
+
+stdenv.mkDerivation rec {
+  pname   = "abuse";
+  version = "0.8";
+
+  src = fetchurl {
+    url    = "http://abuse.zoy.org/raw-attachment/wiki/download/${pname}-${version}.tar.gz";
+    sha256 = "0104db5fd2695c9518583783f7aaa7e5c0355e27c5a803840a05aef97f9d3488";
+  };
+
+  configureFlags = [
+    "--with-x"
+    "--with-assetdir=$(out)/orig"
+    # The "--enable-debug" is to work around a segfault on start, see https://bugs.archlinux.org/task/52915.
+    "--enable-debug"
+  ];
+
+  desktopItems = [ (makeDesktopItem {
+    name = "abuse";
+    exec = "abuse";
+    icon = "abuse";
+    desktopName = "Abuse";
+    comment     = "Side-scroller action game that pits you against ruthless alien killers";
+    categories  = "Game;ActionGame;";
+  }) ];
+
+  postInstall = ''
+    mkdir $out/etc
+    echo -e "dir ${freepats}\nsource ${freepats}/freepats.cfg" > $out/etc/timidity.cfg
+
+    mv $out/bin/abuse $out/bin/.abuse-bin
+    substituteAll "${./abuse.sh}" $out/bin/abuse
+    chmod +x $out/bin/abuse
+
+    install -Dm644 doc/abuse.png $out/share/pixmaps/abuse.png
+  '';
+
+  nativeBuildInputs = [ copyDesktopItems ];
+  buildInputs       = [ SDL SDL_mixer freepats ];
+
+  meta = with stdenv.lib; {
+    description = "Side-scroller action game that pits you against ruthless alien killers";
+    homepage    = "http://abuse.zoy.org/";
+    license     = with licenses; [ publicDomain gpl2Plus wtfpl unfree ];
```

Comment:
How can it be all of these at once?

Reply:
It isn't, I was under the impression that there is no consistent usage of the license field in case more than one license is listed. Sometimes it means "or" (I can pick which license to use, at my discretion), sometimes it means "some parts are such others are such". But I have not seen any written notice on the recommended usage.

Should I change this to just unfree?

Reply:
I downloaded the tarball, and looking at COPYING it appears that they are all free. I would suggestion using `licenses.gpl2Plus`

Reply:
Where did you get that from? I also looked at COPYING. It says:

    data/sfx/*:
    data/music/*:
       Copyright (c) 1995 Bobby Prince <bobby@bpmusic.com>
       Bobby gave Debian permission to redistribute these files.

Also, abuse got removed from Debian from what appears to be licensing issues. (I didn't dig further.) I take this as an indication that abuse is not completely free, at least not in the eyes of the Debian maintainers. What do you think?

Reply:
Could you link to the Debian discussion?

Reply:
Here it is: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=648272

Reply:
Then I suggest marking the package as unfree, as per Debian discussion.

Reply:
Okay, did so. :-) I kinda hoped that I misunderstood the Debian discussion...